### PR TITLE
amp-script: Polish docs, add tutorials links

### PR DESCRIPTION
**Pull request by choumx**
Should we also consolidate these links in the nav bar? Maybe under a single "interactive" section and split along the lines of "combining AMP components with amp-bind" and "writing custom JS".

image

### CODE REVIEW

File: extensions/amp-script/amp-script.md [hunk 1]
```diff
@@ -46,61 +44,101 @@ Allows rendering of custom UI components running on third-party JavaScript.
...
+        <li><a href="https://amp.dev/documentation/guides-and-tutorials/develop/custom-javascript-tutorial?format=websites">Custom password requirements with amp-script</a></li>
+      </ul>
+    </td>
+  </tr>
 </table>
 
 ## Overview
 
-The `amp-script` component allows you to render widgets and other UI using custom third-party JavaScript, e.g. a React component.
+The `amp-script` component allows you run custom JavaScript to render UI e.g. React components.
```
Comment (CrystalOnScript) [inline extensions/amp-script/amp-script.md]: I'd reword as:

he `amp-script` component allows you run custom JavaScript to render UI elements, such as a React component.

File: extensions/amp-script/amp-script.md [hunk 2]
```diff
@@ -46,61 +44,101 @@ Allows rendering of custom UI components running on third-party JavaScript.
...
-`amp-script` is an AMP component wrapper for [`worker-dom`](https://github.com/ampproject/worker-dom/).
+`amp-script` has the following restrictions on JavaScript file size:
 
-`worker-dom` runs third-party JavaScript in a web worker containing a virtual DOM. The virtual DOM listens for mutations and forwards them to the main page which reflects the changes on the real DOM.
+- Maximum of 10,000 bytes per `amp-script` element that uses a local script via `script[type=text/plain][target=amp-script]`.
+- Maximum total of 150,000 bytes for all `amp-script` elements on the page.
 
-For design details, see the ["Intent to Implement" issue](https://github.com/ampproject/amphtml/issues/13471).
-For more information on `worker-dom`, see the [@ampproject/worker-dom](https://github.com/ampproject/worker-dom/) repository.
+#### Required user gestures
```
Comment (CrystalOnScript) [inline extensions/amp-script/amp-script.md]: Are user gestures required? Have we removed the [300px hight exception?](https://amp.dev/documentation/guides-and-tutorials/develop/custom-javascript?format=websites#unprompted-changes)

**Comment by CrystalOnScript** [comment]
@choumx - I think you're onto something with your grouping suggestions. I'm working on an `amp-bind` guide and it would fit well with the custom JS guide under interactivity. However, the two other links are tutorials, and we like to keep those at top level accessible in the nav bar and not nest them with guides. I might rename `create interactive AMP pages` tutorial, as there are now multiple ways to do it. Thanks for adding another pair of eyes!

File: extensions/amp-script/amp-script.md (hunk 2 above)
Comment (choumx) [reply extensions/amp-script/amp-script.md]: It's (3) below: https://github.com/ampproject/amphtml/pull/23221/files#diff-1e9abb643c017d697cacb9e505f22839R145

Changed this to just "User gestures" since it's under "Restrictions" anyways.

File: extensions/amp-script/amp-script.md (hunk 1 above)
Comment (choumx) [reply extensions/amp-script/amp-script.md]: Done.

**Comment by choumx** [comment]
Sounds good, thanks!

File: extensions/amp-script/amp-script.md (hunk 2 above)
Comment (CrystalOnScript) [reply extensions/amp-script/amp-script.md]: Haha, my apologies - I totally just missed this
